pkg/api/validate.go

```diff
@@ -265,7 +268,7 @@ func validateRouterProfile(rp RouterProfile) (errs []error) {
 		errs = append(errs, fmt.Errorf("invalid properties.routerProfiles[%q].publicSubdomain %q", rp.Name, rp.PublicSubdomain))
 	}
 
-	if rp.FQDN != "" && !isValidHostname(rp.FQDN) && !isAzureZone(rp.FQDN) {
+	if rp.FQDN != "" && !isValidCloudAppHostname(rp.FQDN, location) {
```

Comment:
pernicious! this should originally have been `if rp.FQDN != "" && (!isValidHostname(rp.FQDN) || !isAzureZone(rp.FQDN)) {`

Reply:
@thanasisk fyi ^ (also my bad for lgtm'ing it originally)